### DISCUSSION TITLE
Trying to get the CIP service codes from our PLC

### DISCUSSION THREAD

**Discussion by clint70** [dmroeder/pylogix#219] [Q&A]
I am not certain how to query the PLC for the services it has, ie 

    get_attributes_all = b"\x01"
    set_attributes_all = b"\x02"
    get_attribute_list = b"\x03"
    set_attribute_list = b"\x04"
    reset = b"\x05"
    start = b"\x06"
    stop = b"\x07"
    create = b"\x08"
    delete = b"\x09"

Once connected, is there a way to get those details from the PLC or the objects is has?  I am looking for a AB proprietary code, but can't remember.  Thought this would be a good 'educational' moment.  Turns out it is a little more that I was hoping for.

**Comment by dmroeder** [comment]
I am unaware of any way to ask a PLC what services are implemented.  While it seem like a reasonable thing to expect to exist, I have my doubts that something like that does exist. I suspect brute force.  Keep trying services, see what succeeds and fails. 

This isn't the focus of pylogix, so I don't have a way for you to do this.
